### PR TITLE
Cryptography 1.2.3 support

### PR DESCRIPTION
**Pull request by benjaoming**
## Summary

Trying to see if I can implement support for older cryptography versions so they don't mess up when they're present on the host system.

## TODO

- [x] Have tests been written for the new code?
- [x] Has documentation been written/updated?
- [x] New dependencies (if any) added to requirements file

## Reviewer guidance

In progress...

## Issues addressed

#32 #37 

## Documentation

n/a

### CODE REVIEW

**Comment by benjaoming** [comment]
Nah, seems impossible? I cannot find any implementation in cryptography 1.2.3 that supports private key signing of data?

So the ideal would be if Morango ignores cryptography 1.2.3...?

**Comment by benjaoming** [comment]
Very broken on 1.2.3 :)

```
tests/testapp/tests/test_api.py FFFFFFFFFFF...FFFFFFFFFFFFFFFFFFFFFFFFFFFFF.                                                                                               [ 23%]
tests/testapp/tests/test_certificates.py FFFFFFFFFF.                                                                                                                       [ 28%]
tests/testapp/tests/test_connection.py FFFFF.....                                                        
```

**Comment by benjaoming** [comment]
With tests clocking in at 19+ minutes, this clearly needs a very scoped test matrix :) @jamalex @ralphiee22 any opinions?

File: morango/crypto.py [hunk 1]
```diff
@@ -18,6 +18,9 @@
     crypto_backend = default_backend()
     from cryptography.hazmat.primitives.asymmetric import rsa as crypto_rsa, padding as crypto_padding
     from cryptography.hazmat.primitives import serialization as crypto_serialization, hashes as crypto_hashes
+    # Ignore cryptography versions that do not support the 'sign' method
+    if not hasattr(crypto_rsa.RSAPrivateKey, 'sign'):
+        raise ImportError
```
Comment (jamalex) [inline morango/crypto.py]: Interesting that it didn't support this -- seems like a pretty fundamental operation!

Seems like a reasonable approach here. We should perhaps also specify a minversion in the requirements?

**Comment by jamalex** [comment]
We should probably change the requirements to be `>=` rather than `==`?
https://github.com/learningequality/morango/blob/master/requirements/accelerated.txt#L3

And perhaps then also add 1.8.0 to the test matrix?

**Comment by ralphiee22** [comment]
Curious why we want to add 1.8 to the test matrix? b/c it is a LTS version?

**Comment by ralphiee22** [comment]
if we are doing `>=` for the cryptography requirements, can we take out `1.2.3` from the test matrix?

**Comment by benjaoming** [comment]
@ralphiee22 the idea to add `cryptography` to the test matrix was mainly to say that "if present" it's not actually used, but that the tests succeed.

Seeing that they run so incredibly slowly though, I'd prefer to run the tests only for Django 1.11 (currently the Kolibri 0.8 version, and probably going to be the final and only Django version because of Python 2.7)

**Comment by benjaoming** [comment]
I've updated the test matrix such that cryptography 1.2.3 is tested only for Django 1.11 + Python 2.7 and Django 1.11 + Python 3.6

**Comment by benjaoming** [comment]
Also added 1.8.1 with tests for all Python + Django versions (since it's accellerated)

**Comment by benjaoming** [comment]
Sorry about adding such a slow test, could consider to expand the test matrix with build stages: https://blog.travis-ci.com/2017-09-12-build-stages-order-and-conditions

**Comment by benjaoming** [comment]
I cannot grasp the stage rules at this point... am confused about how to include/exclude versions of Python in the matrix.. will just open an issue about it.

Kolibri would benefit a lot from such staged test behavior as well, to separate tests into slow quick-fail tests before running the full set.

**Comment by jamalex** [comment]
Not too worried about morango travis speed at this stage, as we can run the primary test set locally during development. We don't open as many Morango PRs as we do on Kolibri, so it shouldn't be a big slowdown to our flow.

**Comment by benjaoming** [comment]
The latest commit has already been queued for 1 hour.. I think maybe Travis has some way of down-prioritizing these long, slow builds.

**Comment by benjaoming** [comment]
Anyways, if the tests succeed, I think it can be released? @ralphiee22 ?

Ping me if this is released, and we can propose the version bump for Kolibri 0.7.1 immediately :)

**Pull request merged**